File: Python/Algorithm/baekjoon/5430_AC.py
# baekjoon source = "https://www.acmicpc.net/problem/5430"
import sys

T = int(sys.stdin.readline())

for test_case in range(T):
    meto = sys.stdin.readline().strip()
    count = int(sys.stdin.readline())
    len_meto = len(meto)

    first = -1
    c = sys.stdin.readline()
    len_c = len(c)
    c = c[1:len_c - 2].split(',')
    rear = count - 1

    if count == 0:
        if 'D' in meto:
            print('error')
        else:
            print('[]')
    else:
        flag = True
        for i in range(len_meto):
            if meto[i] == 'D':
                if first == rear:
                    print('error')
                    break
                if flag:
                    first += 1
                    c[first] = 0
                else:
                    c[rear] = 0
                    rear -= 1
            else:
                if flag:
                    flag = False
                else:
                    flag = True
        else:
            print('[', end='')
            if flag:
                for j in range(first + 1, rear + 1):
                    if j != rear:
                        print(c[j] + ',', end='')
                    else:
                        print(c[j], end='')
                print(']')
            else:
                for j in range(rear, first, -1):
                    if j != first+1:
                        print(c[j] + ',', end='')
                    else:
                        print(c[j], end='')
                print(']')

File: Java/baekjoon/화목한스터디/src/큐/_5430_AC.java
//baekjoon source = "https://www.acmicpc.net/problem/5430"
package ť;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Collections;
import java.util.Deque;
import java.util.LinkedList;
import java.util.List;
import java.util.StringTokenizer;

public class _5430_AC {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		int T = Integer.parseInt(br.readLine());
		for (int t = 0; t < T; t++) {
			char[] method;
			StringTokenizer st;
			st = new StringTokenizer(br.readLine());
			method = st.nextToken().toCharArray();
			Deque<String> deq = new LinkedList<String>();
			st = new StringTokenizer(br.readLine());
			st.nextToken();
			st = new StringTokenizer(br.readLine());
			String[] s = st.nextToken().replace('[', ' ').replace(']', ' ').trim().split(",");
			for (int i = 0; i < s.length; ++i) {
				if (s[i].equals("")) continue;
				deq.add(s[i]);
			}

			StringBuilder sb = new StringBuilder();
			boolean flag = true;

			for (int i = 0; i < method.length; i++) {
				if (method[i] == 'R') {
					flag = !flag;
				} else {
					if (deq.isEmpty()) {
						sb.append("error");
						break;
					}
					if (flag) {
						deq.pollFirst();
					} else {
						deq.pollLast();
					}
				}
			}
			if (!flag) {
				Collections.reverse((List<?>) deq);
			}
			if (sb.length() == 0) {
				sb.append('[');
				while(!deq.isEmpty()) {
					sb.append(deq.pollFirst());
					if (deq.size() != 0) {
						sb.append(',');
					}
				}
				sb.append(']');
			}
			bw.write(sb + "\n");
			bw.flush();
		}
	}
}
